### PR TITLE
chore(agent,sysdig-deploy): set promscrape port to 9091 by default when gke autopilot is enabled

### PR DESCRIPTION
**Pull request by francesco-furlan**
## Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [X] Title of the PR starts with type and scope, (e.g. feat(agent,node-analyzer,sysdig-deploy):)
- [X] Chart Version bumped for the respective charts
- [ ] Variables are documented in the README.md (or README.tpl in some charts)
- [X] Check GithubAction checks (like lint) to avoid merge-check stoppers
- [X] All test files are added in the tests folder of their respective chart and have a "_test" suffix

Check Contribution guidelines in README.md for more insight.

### CODE REVIEW

File: charts/sysdig-deploy/Chart.yaml
Comment (aroberts87) [inline charts/sysdig-deploy/Chart.yaml:1]: Hey @francesco-furlan! This step will automatically happen automatically via CI actions when you merge a PR that bumps the patch version of a dependency of `sysdig-deploy`. It doesn't hurt to keep it here, but you can omit it from future PRs if you wish.

File: charts/agent/templates/configmap-deployment.yaml [hunk 1]
```diff
@@ -47,6 +47,9 @@ data:
   If these conditions are true, then we use the agent.sslCaFileName for the http_proxy.ca_certificate
 */}}
 {{- $baseSettings := .Values.sysdig.settings -}}
+{{- if and (include "agent.gke.autopilot" .) (or (not (hasKey $baseSettings "promscrape_web_address")) (eq (get $baseSettings "promscrape_web_address") "127.0.0.1:9990") ) }}
+  {{- $baseSettings := mergeOverwrite $baseSettings (dict "promscrape_web_address" "127.0.0.1:9991") -}}
+{{- end }}
```
Comment (aroberts87) [inline charts/agent/templates/configmap-deployment.yaml]: This should not be needed. `prometheus.enabled` is set to `false` earlier in the configuration, and promscrape will not run in that situation. 
cc @thom-sd

File: charts/sysdig-deploy/Chart.yaml
Comment (francesco-furlan) [reply charts/sysdig-deploy/Chart.yaml:1]: yep I know, but there's something in my head that tells me to do it everytime by hand 😅 😂

File: charts/agent/templates/configmap-deployment.yaml (hunk 1 above)
Comment (francesco-furlan) [reply charts/agent/templates/configmap-deployment.yaml]: nice catch, thanks!

File: charts/agent/tests/gke_test.yaml [hunk 2]
```diff
@@ -106,6 +106,45 @@ tests:
...
+      gke:
+        autopilot: true
+    asserts:
+      - matchRegex:
+          path: data['dragent.yaml']
+          pattern: |-
+            promscrape_web_address: 127.0.0.1:9991
+    template: templates/configmap.yaml
+
+  - it: Ensure promscrape_web_ddress is unaltered when is set on settings and autopilot is enabled
```
Comment (fuster92) [inline charts/agent/tests/gke_test.yaml]:
```suggestion
  - it: Ensure promscrape_web_address is unaltered when is set on settings and autopilot is enabled
```
Small typo 🙈

**Review by fuster92 — COMMENTED** [review]
LGTM

**Review by aroberts87 — APPROVED** [review]

**Pull request merged**